codingstandards/repositorynamingconvention.md

```diff
@@ -0,0 +1,196 @@
+# Repository Naming Convention
+All data access should be done through a repository class.  These classes should follow a specific
+naming convention to help callers better understand the functionality available.
+
+## Interface Definition
+Any repository class should be implementing an interface.  This allows the callers to use dependency
+injection with more test-friendly implementations.  The interface should be named with the "I" prefix according
+to the coding standard, followed by the primary entity that it manages for its CRUD operations.
+
+## Method Names
+Methods for CRUD operations follow very consistent patterns, so it is extremely helpful to the callers of
+our repository classes to have consistent names, indicating what the behavior is exactly.
+
+Below is an example to reference as we explain the naming convention.
+
+    // Interface for the persistence layer of orders.
```

Comment:
You can give type hinting in here by doing a triple tick and the type, so in this case, "```csharp" followed by your code, and ending in a triple tick.

```csharp
// Interface test..
public interface IOrder
{
}
```